File: dataset/Vanilla - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task25.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char** number(char** lines, int size, int* resultSize) {
    *resultSize = size;
    if (size == 0) {
        return NULL;
    }
    
    char** result = (char**)malloc(size * sizeof(char*));
    for (int i = 0; i < size; i++) {
        int numDigits = snprintf(NULL, 0, "%d", i + 1);
        int totalLen = numDigits + 2 + strlen(lines[i]) + 1; // number + ": " + line + '\\0'
        result[i] = (char*)malloc(totalLen * sizeof(char));
        sprintf(result[i], "%d: %s", i + 1, lines[i]);
    }
    return result;
}

void freeResult(char** result, int size) {
    for (int i = 0; i < size; i++) {
        free(result[i]);
    }
    free(result);
}

int main() {
    int resultSize;
    
    // Test case 1: Empty list
    char** test1 = NULL;
    char** result1 = number(test1, 0, &resultSize);
    printf("Test 1: []\\n");
    
    // Test case 2: Basic example
    char* test2[] = {"a", "b", "c"};
    char** result2 = number(test2, 3, &resultSize);
    printf("Test 2: [");
    for (int i = 0; i < resultSize; i++) {
        printf("\\"%s\\"", result2[i]);
        if (i < resultSize - 1) printf(", ");
    }
    printf("]\\n");
    freeResult(result2, resultSize);
    
    // Test case 3: Single element
    char* test3[] = {"hello"};
    char** result3 = number(test3, 1, &resultSize);
    printf("Test 3: [");
    for (int i = 0; i < resultSize; i++) {
        printf("\\"%s\\"", result3[i]);
        if (i < resultSize - 1) printf(", ");
    }
    printf("]\\n");
    freeResult(result3, resultSize);
    
    // Test case 4: Multiple words
    char* test4[] = {"hello world", "foo bar", "test"};
    char** result4 = number(test4, 3, &resultSize);
    printf("Test 4: [");
    for (int i = 0; i < resultSize; i++) {
        printf("\\"%s\\"", result4[i]);
        if (i < resultSize - 1) printf(", ");
    }
    printf("]\\n");
    freeResult(result4, resultSize);
    
    // Test case 5: Empty strings
    char* test5[] = {"", "a", ""};
    char** result5 = number(test5, 3, &resultSize);
    printf("Test 5: [");
    for (int i = 0; i < resultSize; i++) {
        printf("\\"%s\\"", result5[i]);
        if (i < resultSize - 1) printf(", ");
    }
    printf("]\\n");
    freeResult(result5, resultSize);
    
    return 0;
}

File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task25.java

import java.util.ArrayList;
import java.util.List;

public class Task25 {
    public static List<String> number(List<String> lines) {
        if (lines == null) {
            return new ArrayList<>();
        }
        
        List<String> result = new ArrayList<>();
        for (int i = 0; i < lines.size(); i++) {
            String line = lines.get(i);
            if (line == null) {
                line = "";
            }
            result.add((i + 1) + ": " + line);
        }
        return result;
    }
    
    public static void main(String[] args) {
        // Test case 1: Empty list
        System.out.println("Test 1: " + number(new ArrayList<>()));
        
        // Test case 2: Basic example
        List<String> test2 = new ArrayList<>();
        test2.add("a");
        test2.add("b");
        test2.add("c");
        System.out.println("Test 2: " + number(test2));
        
        // Test case 3: Single element
        List<String> test3 = new ArrayList<>();
        test3.add("hello");
        System.out.println("Test 3: " + number(test3));
        
        // Test case 4: Strings with special characters
        List<String> test4 = new ArrayList<>();
        test4.add("Hello, World!");
        test4.add("Line with spaces");
        test4.add("Special @#$%");
        System.out.println("Test 4: " + number(test4));
        
        // Test case 5: Empty strings
        List<String> test5 = new ArrayList<>();
        test5.add("");
        test5.add("non-empty");
        test5.add("");
        System.out.println("Test 5: " + number(test5));
    }
}
